gateway/src/test/java/io/camunda/zeebe/gateway/api/job/FailJobTest.java

```diff
@@ -54,4 +57,45 @@ public void shouldMapRequestAndResponse() {
     assertThat(brokerRequestValue.getRetryBackoff()).isEqualTo(retryBackOff);
     assertThat(brokerRequestValue.getErrorMessageBuffer()).isEqualTo(wrapString("failed"));
   }
+
+  @Test
+  public void shouldSetVariablesLocally() {
+    // given
+    final FailJobStub stub = new FailJobStub();
+    stub.registerWith(brokerClient);
+
+    final int retries = 123;
+    final int retryBackOff = 100;
+
+    final String variables = JsonUtil.toJson(Collections.singletonMap("foo", "bar"));
+
+    final FailJobRequest request =
+        FailJobRequest.newBuilder()
+            .setJobKey(stub.getKey())
+            .setRetries(retries)
+            .setRetryBackOff(retryBackOff)
+            .setErrorMessage("failed")
+            .setVariables(variables)
+            .build();
+
+    // when
+    final FailJobResponse response = client.failJob(request);
+
+    // then
+    assertThat(response).isNotNull();
+
+    final BrokerFailJobRequest brokerRequest = brokerClient.getSingleBrokerRequest();
+
+    assertThat(brokerRequest.getKey()).isEqualTo(stub.getKey());
+    assertThat(brokerRequest.getIntent()).isEqualTo(JobIntent.FAIL);
+    assertThat(brokerRequest.getValueType()).isEqualTo(ValueType.JOB);
+
+    final JobRecord brokerRequestValue = brokerRequest.getRequestWriter();
+    assertThat(brokerRequestValue.getRetries()).isEqualTo(retries);
+    assertThat(brokerRequestValue.getRetryBackoff()).isEqualTo(retryBackOff);
+    assertThat(brokerRequestValue.getErrorMessageBuffer()).isEqualTo(wrapString("failed"));
+    MsgPackUtil.assertEqualityExcluding(brokerRequestValue.getVariablesBuffer(), variables);
+
+    assertThat(brokerRequestValue.getVariables().get("foo")).isEqualTo("bar");
```

Comment:
🔧 This does not have to be a new test case, it can simply be added to the existing `shouldMapRequestAndResponse`

Reply:
Done.